filters/auth/oidc.go

```diff
@@ -368,10 +387,10 @@ func extractDomainFromHost(host string) string {
 	if ip != nil {
 		return ip.String()
 	}
-	if strings.Count(h, ".") < 2 {
+	if 1+strings.Count(h, ".")-subdomainsToRemove < 2 {
 		return h
 	}
-	return strings.Join(strings.Split(h, ".")[1:], ".")
+	return strings.Join(strings.Split(h, ".")[subdomainsToRemove:], ".")
```

Comment:
There is no need split if there nothing to remove and there is no need to count if we split anyway:
```go
if subdomainsToRemove == 0 {
    return h
}
subDomains := strings.Split(h, ".")
if len(subDomains) - subdomainsToRemove < 2 {
    return h
}
return strings.Join(subDomains[subdomainsToRemove:], ".")
```